code/modules/modular_computers/laptop_vendor.dm

```diff
@@ -261,103 +254,109 @@
 		ui = new(user, src, ui_key, "computer_fabricator.tmpl", "Personal Computer Vendor", 500, 400)
 		ui.set_initial_data(data)
 		ui.open()
-		ui.set_auto_update(1)
+		ui.set_auto_update(TRUE)
 
 /obj/machinery/lapvend/attackby(obj/item/W, mob/user)
 	if(W.iswrench())
 		user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
-		playsound(src.loc, W.usesound, 100, 1)
+		playsound(get_turf(src), W.usesound, 100, TRUE)
 		if(anchored)
-			user.visible_message("[user] begins unsecuring \the [src] from the floor.", "You start unsecuring \the [src] from the floor.")
+			user.visible_message(SPAN_NOTICE("\The [user] begins unsecuring \the [src] from the floor."), SPAN_NOTICE("You start unsecuring \the [src] from the floor."))
 		else
-			user.visible_message("[user] begins securing \the [src] to the floor.", "You start securing \the [src] to the floor.")
-		if(do_after(user, 20/W.toolspeed))
+			user.visible_message(SPAN_NOTICE("\The [user] begins securing \the [src] to the floor."), SPAN_NOTICE("You start securing \the [src] to the floor."))
+		if(do_after(user, 20 / W.toolspeed))
 			if(!src)
 				return
-			to_chat(user, span("notice", "You [anchored ? "un" : ""]secured \the [src]!"))
+			to_chat(user, SPAN_NOTICE("You [anchored ? "un" : ""]secured \the [src]!"))
 			anchored = !anchored
 		return
+	else if(state != 2 && (istype(W, /obj/item/card/id) || istype(W, /obj/item/device/pda) || istype(W, /obj/item/modular_computer) || istype(W, /obj/item/card/tech_support)))
+		to_chat(user, SPAN_WARNING("\The [src] isn't in the payment mode yet!"))
```

Comment:
Bit too gamey. Could just tell them that they're having trouble slamming their ID into a slot that's not open :^)